Products/ZenUI3/browser/eventconsole/export.py

```diff
@@ -75,11 +77,19 @@ def csv(self, response, archive, **params):
         response.setHeader('Content-Disposition', 'attachment; filename=events.csv')
         from csv import writer
         writer = writer(response)
-
+        events = [dict((k, v) for k, v in evt.iteritems() if v) for _, evt in
+                  self._query(archive, **params)]
+        keys = [x.keys() for x in events]
+        fields = {item for sublist in keys for item in sublist}
```

Comment:
I'm not sure whether this is any faster (or slower), but is probably a little more readable:
```
fields = {item for item in itertools.chain.from_iteratable(keys)}
```
Just remember to `import itertools`